src/main/scala/abalone/format/FEN.scala

```diff
@@ -1,33 +1,60 @@
 package strategygames.abalone.format
 
 import strategygames.Player
+import strategygames.abalone.File
+import strategygames.abalone.Piece
 import strategygames.abalone.PieceMap
+import strategygames.abalone.Pos
+import strategygames.abalone.Stone
 
 final case class FEN(value: String) extends AnyVal {
+  // squares are described from topLeft to bottomRight in the FEN :
+  // eg. Belgian Daisy: SS1ss/SSSsss/1SS1ss1/8/9/8/1ss1SS1/sssSSS/ss1SS
+  // Snakes: sssss/s5/s6/s1SSSSS1/1s5S1/1sssss1S/6S/5S/SSSSS
 
   override def toString = value
 
-  def player: Option[Player] =
-    value.split(' ').lift(3) flatMap (_.headOption) flatMap Player.apply
+  def pieces: PieceMap = squares.zipWithIndex.map {
+      case (pieces, row) => pieces.zipWithIndex.map[Option[(Option[strategygames.abalone.Pos], strategygames.abalone.Piece)]] { 
```

Comment:
This code looks a bit bizarre to me.

Firstly, I try to avoid doing full path calls to objects in the actual code (unless you need to be explicit, e.g. in wrapper layer). As you've imported `strategygames.abalone.Pos` you should be able to just refer to `Pos` here. Same with `Piece` and `P1` should be accessible from `Player.P1` or just import P1 and P2.

Then, I'm not sure you need to explicitly write the type `Option[(Option[strategygames.abalone.Pos], strategygames.abalone.Piece)]` but maybe you do.

Thirdly the way I read this code, you are taking `squares` and doing `zipWithIndex` and producing `(pieces, row)` which is essentially `(squares, index)`. And then you are doing `zipWithIndex` on `pieces` again which is really just `squares` so you are redoing the same zip it seems to me?

Reply:
this is due to the fact I do not know how to code in Scala ;)
good to be able to have a look together tomorrow